### DISCUSSION TITLE
sonoff ZBdongle-P  not showing in texas instruments FLASH-PROGRAMMER-2

### DISCUSSION THREAD

**Discussion by kokomatamata** [Koenkk/Z-Stack-firmware#504] [Q&A]
I attempted to flash a sonoff ZBdongle-P (Texas Instruments CC2652P) with the zigbee2mqtt suggested firmware. When connected to a windows 11 pc, the ZBdongle-P was not recognized by the installed  texas instruments FLASH-PROGRAMMER-2. I attempted to place the dongle in bootloader mode several times. The device manager in windows tells me that a driver is missing.  Is a windows driver necessary so that the FLASH-PROGRAMMER-2 is able to access the dongle? If yes where can I get it?

**Comment by propi62** [comment]
the easiest way to flash sonof dongle P for me on windows is: https://github.com/xyzroe/ZigStarGW-MT
there is also another way to flash it directly within Homeassistant addon.

no bootloader button press (unscrewing) is needed.
